pkg/tools/otelcol-config/delete_tag.go

```diff
@@ -0,0 +1,97 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"sort"
+
+	"github.com/mikefarah/yq/v4/pkg/yqlib"
+)
+
+// DeleteTagAction deletes a collector tag from conf.d.
+// If the --override flag is present, it will delete the tag from both the
+// overrides file and the settings file. If the --override flag is present and
+// the tag exists in a user-controlled file, an error will be returned.
+func DeleteTagAction(ctx *actionContext) error {
+	conf, err := ReadConfigDir(ctx.ConfigDir)
+	if err != nil {
+		return err
+	}
+
+	if conf.SumologicRemote != nil {
+		return errors.New("delete-tag not supported for remote-controlled collectors")
+	}
+
+	if ctx.Flags.Override {
+		return deleteTagOverride(ctx, conf)
+	} else {
+		return deleteTag(ctx, conf, ConfDSettings, ctx.WriteConfD)
+	}
+}
+
+func deleteTagOverride(ctx *actionContext, conf ConfDir) error {
+	readOrder := sort.StringSlice{}
+	for key := range conf.ConfD {
+		readOrder = append(readOrder, key)
+	}
+	sort.Sort(sort.Reverse(readOrder))
```

Comment:
I'm not 100% sure why we need to sort here?

Reply:
https://help.sumologic.com/docs/send-data/opentelemetry-collector/data-source-configurations/overview/#configuration-structure 

```
If there are any conflicts between the files, the last loaded configuration file will take precedence.
```

i think its so it gets the last loaded config? 🤔

Reply:
It's actually not strictly required to visit these files in order in this specific case, this is copypasta from read-kv :laughing:

Reply:
But yes, there is an order of precedence that needs to be considered here sometimes.

Reply:
I think I'm going to leave it sorted, so that errors are stable on subsequent runs in the case of there being more than one error in the corpus.